src/functions-reference/compound_arithmetic_and_assignment.Rmd

```diff
@@ -29,274 +11,106 @@ cat(' * <a href="compound-elementwise-division-and-assignment.html">Compound Ele
 }
 ```
 
-## Compound addition and assignment
-
-<!-- void; operator+=; (int x, int y); -->
-\index{{\tt \bfseries operator\_compound\_add }!{\tt (int x, int y): void}|hyperpage}
-
-`void` **`operator+=`**`(int x, int y)`<br>\newline
-`x += y` is equivalent to `x = x + y`.
-`r since("2.17")`
-
-<!-- void; operator+=; (real x, real y); -->
-\index{{\tt \bfseries operator\_compound\_add }!{\tt (real x, real y): void}|hyperpage}
-
-`void` **`operator+=`**`(real x, real y)`<br>\newline
-`x += y` is equivalent to `x = x + y`.
-`r since("2.17")`
-
-<!-- void; operator+=; (vector x, real y); -->
-\index{{\tt \bfseries operator\_compound\_add }!{\tt (vector x, real y): void}|hyperpage}
+Compound arithmetic and assignment statements combine an arithmetic
+operation and assignment, replacing a statement such as
 
-`void` **`operator+=`**`(vector x, real y)`<br>\newline
-`x += y` is equivalent to `x = x + y`.
-`r since("2.17")`
+```
+ x = x op y;
+```
 
-<!-- void; operator+=; (row_vector x, real y); -->
-\index{{\tt \bfseries operator\_compound\_add }!{\tt (row\_vector x, real y): void}|hyperpage}
+with the more compact compound form
 
-`void` **`operator+=`**`(row_vector x, real y)`<br>\newline
-`x += y` is equivalent to `x = x + y`.
-`r since("2.17")`
+```
+ x op= y;
+```
 
-<!-- void; operator+=; (matrix x, real y); -->
-\index{{\tt \bfseries operator\_compound\_add }!{\tt (matrix x, real y): void}|hyperpage}
+For example, `x = x + 1;` may be replaced with `x += 1;`.  This works
+for all types that support arithmetic, including the scalar
+types `int`, `real`, `complex`, the real matrix types `vector`,
+`row_vector`, and `matrix`, and the complex matrix types,
+`complex_vector`, `complex_row_vector`, and `complex_matrix`.
 
-`void` **`operator+=`**`(matrix x, real y)`<br>\newline
-`x += y` is equivalent to `x = x + y`.
-`r since("2.17")`
 
-<!-- void; operator+=; (vector x, vector y); -->
-\index{{\tt \bfseries operator\_compound\_add }!{\tt (vector x, vector y): void}|hyperpage}
+## Compound addition and assignment
 
-`void` **`operator+=`**`(vector x, vector y)`<br>\newline
-`x += y` is equivalent to `x = x + y`.
-`r since("2.17")`
+Compound addition and assignment works wherever the corresponding
+addition and assignment would be well formed.  
 
-<!-- void; operator+=; (row_vector x, row_vector y); -->
-\index{{\tt \bfseries operator\_compound\_add }!{\tt (row\_vector x, row\_vector y): void}|hyperpage}
+<!-- void; operator+=; (T x, U y); -->
+\index{{\tt \bfseries operator\_compound\_add }!{\tt (T x, U y): void}|hyperpage}
 
-`void` **`operator+=`**`(row_vector x, row_vector y)`<br>\newline
-`x += y` is equivalent to `x = x + y`.
+`void` **`operator+=`**`(T x, U y)`<br>\newline
+`x += y` is equivalent to `x = x + y`.  Defined for all types `T` and `U`
+where `T = T + U` is well formed. Complex signatures added in version 2.30.
 `r since("2.17")`
```

Comment:
I definitely like how this section is organized now. As a note, in other places where we had a change later, we include it in the `since()` command. e.g.,

```
`r since("2.17, complex signatures added in 2.30")`
```

Reply:
I updated to that form.  I didn't do that initially because I didn't know if the numbers were getting used programatically somewhere and I didn't want to break anything.